server.go

```diff
@@ -93,6 +93,28 @@ func (oa *onionAddr) Network() string {
 // Ensure onionAddr implements the net.Addr interface.
 var _ net.Addr = (*onionAddr)(nil)
 
+// onionAddr implements the net.Addr interface with two struct fields
+type simpleAddr struct {
+	net, addr string
+}
+
+// String returns the address.
+//
+// This is part of the net.Addr interface.
+func (a *simpleAddr) String() string {
```

Comment:
If you change this to be a value receiver, it will allow it work with anything that uses a `fmt.Stringer` interface (e.g. `fmt.Printf`, `log.Warnf`, etc) so don't need to modify the logging statements.

As an aside, `fmt.Stringers` should pretty much always be value receives because Go will automatically deference pointers, so it works with both.

Example:
```Go
package main

import(
	"net"
	"fmt"
)

type foo struct {
	addr string
}

func (f foo) String() string {
	return f.addr
}

func (f foo) Network() string {
	return "moo"
}


func main(){
	addrs := []net.Addr{&foo{"1"}, foo{"2"}}
	for _, addr := range addrs {
		fmt.Println(addr)
	}
}
```
Output:
```
1
2
```